File: app/models/tile_type.rb
class TileType < ActiveRecord::Base
  validates :name, :presence=>true
  validates :img, :presence=>true
  validates :img_x, :presence=>true, :numericality => {:greater_than_or_equal_to => 0}
  validates :img_y, :presence=>true, :numericality => {:greater_than_or_equal_to => 0}
  acts_as_list
  has_many :terrain_modifiers
  
  def unit_types_available=(val)
    super(val.keys.join(','))
  end
  
  def unit_types_available
    if super()
      super().split(',')
    else
      []
    end
  end
end






# == Schema Information
#
# Table name: tile_types
#
#  id                   :integer         not null, primary key
#  name                 :string(255)
#  img                  :string(255)
#  position             :integer
#  created_at           :datetime
#  updated_at           :datetime
#  img_x                :integer         default(0), not null
#  img_y                :integer         default(0), not null
#  ownable              :boolean
#  is_store             :boolean         default(FALSE), not null
#  unit_types_available :string(255)
#

File: app/assets/javascripts/models/tile_type.js
namespace('Hexwar.TileType');

/**
 * Represents a type of map tile
 * @constructor
 */
Hexwar.TileType = function (name, img, img_x, img_y, ownable, is_store, unit_types_available) {
	this.name = name;
	this.ownable = ownable;
	this.img = { src: img, x: img_x, y:img_y };
	this.is_store = is_store
	this.unit_types_available = unit_types_available;
}
